Set a non-zero AAGUID for platform credentials where attestation=none

pascoej opened the pull request:
#### 711e9d271fec39992f5febf9d1968fe45b1a955d
<pre>
Set a non-zero AAGUID for platform credentials where attestation=none
<a href="https://bugs.webkit.org/show_bug.cgi?id=261340">https://bugs.webkit.org/show_bug.cgi?id=261340</a>
rdar://115125044

Reviewed by Brent Fulgham.

The other FIDO platform providers return non-zero values in
this case, making us unique by having all zeros.

This patch assigns a random, fixed, AAGUID instead.

* LayoutTests/http/wpt/webauthn/public-key-credential-create-success-local.https.html
* Source/WebCore/Modules/webauthn/WebAuthenticationUtils.cpp
* Source/WebCore/Modules/webauthn/WebAuthenticationUtils.h
* Source/WebCore/Modules/webauthn/fido/DeviceResponseConverter.cpp
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalConnection.h
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalConnection.mm
* Source/WebKit/UIProcess/WebAuthentication/Mock/MockLocalConnection.h
* Source/WebKit/UIProcess/WebAuthentication/Mock/MockLocalConnection.mm
* Source/WebKit/UIProcess/WebAuthentication/Virtual/VirtualLocalConnection.h
* Source/WebKit/UIProcess/WebAuthentication/Virtual/VirtualLocalConnection.mm

Canonical link: <a href="https://commits.webkit.org/269536@main">https://commits.webkit.org/269536@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e91d73cf35ddea1def0642d457981b3a1c25b7c4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22704 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/378 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23791 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24611 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21019 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/22963 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/1453 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23234 "Built successfully") | [⏳ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/WPE-WK2-Tests-EWS "Waiting to run tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22944 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/228 "Passed tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25464 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20569 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26802 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20601 "Passed tests") | [⏳ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24642 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/253 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18101 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/196 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/20372 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5439 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/324 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/258 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/Modules/webauthn/WebAuthenticationUtils.cpp:
```diff
@@ -148,9 +148,11 @@ cbor::CBORValue::MapValue buildAttestationMap(Vector<uint8_t>&& authData, String
 }
 
-Vector<uint8_t> buildAttestationObject(Vector<uint8_t>&& authData, String&& format, cbor::CBORValue::MapValue&& statementMap, const AttestationConveyancePreference& attestation)
+
```
Nit: Extra whitespace!

brentfulgham commented on Source/WebCore/Modules/webauthn/fido/DeviceResponseConverter.cpp:
```diff
@@ -145,7 +145,7 @@ RefPtr<AuthenticatorAttestationResponse> readCTAPMakeCredentialResponse(const Ve
         // is that they are CBORValue instead of the raw type.
         // Also, format and attStmt are omitted as they are not useful in none attestation.
-        attestationObject = buildAttestationObject(Vector<uint8_t>(authenticatorData.getByteString()), String { emptyString() }, { }, attestation);
+        attestationObject = buildAttestationObject(Vector<uint8_t>(authenticatorData.getByteString()), String { emptyString() }, { }, attestation, true /* shouldZeroAAGUID */);
```
Consider making a bool-backed enum to avoid having to make these little comments!

brentfulgham commented on Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm:
```diff
@@ -546,29 +550,14 @@ static inline uint8_t authDataFlags(ClientDataType type, LocalConnection::UserVe
-        return;
-    }
-
-    // Step 13. Apple Attestation
```
This makes me sad. :-(

brentfulgham commented on Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalConnection.mm:
```diff
@@ -199,15 +199,6 @@ static inline String bundleName()
 
-void LocalConnection::getAttestation(SecKeyRef privateKey, NSData *authData, NSData *hash, AttestationCallback&& completionHandler) const
-{
-#if HAVE(APPLE_ATTESTATION)
```
Should we be removing the various Apple Attestation methods, soft-link, etc.?

brentfulgham reviewed (approved):
Looks good. r=me

brentfulgham reviewed (approved):
This looks fine, though I'm sad to see the attestation feature be removed. r=me

pascoej committed:
Set a non-zero AAGUID for platform credentials where attestation=none
https://bugs.webkit.org/show_bug.cgi?id=261340
rdar://115125044

Reviewed by Brent Fulgham.

The other FIDO platform providers return non-zero values in
this case, making us unique by having all zeros.

This patch assigns a random, fixed, AAGUID instead.

* LayoutTests/http/wpt/webauthn/public-key-credential-create-success-local.https.html
* Source/WebCore/Modules/webauthn/WebAuthenticationUtils.cpp
* Source/WebCore/Modules/webauthn/WebAuthenticationUtils.h
* Source/WebCore/Modules/webauthn/fido/DeviceResponseConverter.cpp
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalConnection.h
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalConnection.mm
* Source/WebKit/UIProcess/WebAuthentication/Mock/MockLocalConnection.h
* Source/WebKit/UIProcess/WebAuthentication/Mock/MockLocalConnection.mm
* Source/WebKit/UIProcess/WebAuthentication/Virtual/VirtualLocalConnection.h
* Source/WebKit/UIProcess/WebAuthentication/Virtual/VirtualLocalConnection.mm

Canonical link: https://commits.webkit.org/269536@main
